app/controllers/divisions_controller.rb

```diff
@@ -8,29 +8,34 @@ def index_redirect
   end
 
   def index
+    @years = (Division.order(:date).first.date.year..Division.order(:date).last.date.year).to_a
+
     if params[:mpc] && params[:mpn]
-      electorate = params[:mpc].gsub("_", " ")
-      name = params[:mpn].gsub("_", " ")
+      @mpc = params[:mpc]
+      @mpn = params[:mpn]
+      @house = params[:house]
 
-      @member = Member.with_name(name)
-      @member = @member.in_house(params[:house])
-      @member = @member.where(constituency: electorate)
-      @member = @member.order(entered_house: :desc).first
+      @member = get_member
```

Comment:
I liked this specific block of refactoring, if people like it I will do similar refactorings with the rest of this controller